openadapt/signals.py

```diff
@@ -0,0 +1,289 @@
+import requests
+import importlib
+import os
+import mimetypes
+import sys #for debugging
+import sqlite3
+
+from loguru import logger
+
+
+class Signals:
+    """
+    Create Signal structure as dictionary
+    signals = [{},{},{}]
+
+    {
+        "number": signal number in list,
+        "title": given signal title,
+        "type": type of signal (file, database, etc.)
+        "address": given signal address,
+        "description": description of signal (file type, length, etc.),
+    }
+    """
+    def __init__(self):
+        self.signals = []
+
+    def return_signals(self):
+        """
+        Return the list of signals.
+        """
+        return self.signals
+    
+
+    def __setup_database_signal(self, db_url):
+        """
+        Read a description of a signal from a database.
+        """
+        # Get the signal from the database.
+        #TODO: implement database signal
+
+        conn = sqlite3.connect(db_url)
+        cur = conn.cursor()
+        cur.execute("SELECT * FROM sqlite_master WHERE type='table';")
+
+        rows = cur.fetchall()
+        conn.close()
+
+        result = {}
+
+        for row in rows:
+            result[row[0]] = row[1]
+
+        # Prepare the result
+        table_info = []
+
+        for row in rows:
+            # The first element is the table name, the second element is the SQL schema
+            table_info.append(f"Table: {row[0]}, Schema: {row[1]}")
```

Comment:
What do you think about storing these as state variables on the Signal?

Reply:
Thanks for the advice, for now I believe we'll stick to using if else statements when retrieving metadata or data. Though signal elements will be stored directly instead of converting them to a string first.

Reply:
> Though signal elements will be stored directly instead of converting them to a string first.

Implemented in [0b10c77](https://github.com/OpenAdaptAI/OpenAdapt/pull/160/commits/0b10c7737ae07956248f3d326641c64be1e96bbc)